### PR TITLE
added footer component to layout.

### PR DESCRIPTION
**Pull request by rotegras**
links + footer logo + layout.

### CODE REVIEW

File: src/components/Footer/Footer.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import Styled from 'styled-components';
+// import { ReactComponent as FooterLogo } from '../Header/logo.svg';
```
Comment (jkettmann) [inline src/components/Footer/Footer.js]: Remove comment

File: src/components/Footer/Footer.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    margin-left: 55px;
+  }
+`;
+
+
+export default function Footer() {
+  return (
+    <StyledFooter>
+      <a
+        className="footer-link"
```
Comment (jkettmann) [inline src/components/Footer/Footer.js]: Don't use global classes. Rather create a separate styled component `FooterLink`. Same goes for the links below

File: src/components/Footer/Footer.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        href="https://www.ooloo.io"
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        ooloo.io
+      </a>
+      <a className="footer-link footer-logo" href="/">
```
Comment (jkettmann) [inline src/components/Footer/Footer.js]: This can be a separate styled component since it doesn't need to share styles with the other links

File: src/App.js [hunk 4]
```diff
@@ -6,14 +6,18 @@ import {
...
 import Header from './components/Header/Header';
 import Search from './pages/Search';
 import HowItWorks from './pages/HowItWorks';
 
 const StyledWrapper = Styled.div`
+    position: relative;
+    width: 100%;
     max-width: 1440px;
     margin: 0 auto;
+    overflow-x: hidden;
```
Comment (jkettmann) [inline src/App.js]: Why did you add this? If something is overflowing it might be a problem in the CSS

File: src/GlobalStyle.js [hunk 5]
```diff
@@ -8,21 +8,24 @@ const GlobalStyle = createGlobalStyle`
...
   }
 
   html,
   body {
-    min-height: 100vh;
+    height: 100%;
+    width: 100%;
   }
 
   .page {
```
Comment (jkettmann) [inline src/GlobalStyle.js]: This shouldn't be a global class. You can create a separate component `Page` or similar and use that as a wrapper component for your pages.

File: src/GlobalStyle.js [hunk 6]
```diff
@@ -8,21 +8,24 @@ const GlobalStyle = createGlobalStyle`
...
     display: flex;
     flex-direction: column;
-    justify-content: center;
+    justify-content: flex-start;
     align-items: center;
     flex-grow: 1;
-    padding-top: 22px;
+    position: relative;
+    min-height: 100vh;
```
Comment (jkettmann) [inline src/GlobalStyle.js]: Having a min-height is a good idea. But the complete page should have min-height 100vh, so here it should be 100vh - headerHeight - footerHeight.

I guess it would be good if you create a [theme](https://styled-components.com/docs/advanced#theming) where you can store these heights, colors and so on

File: src/GlobalStyle.js [hunk 7]
```diff
@@ -8,21 +8,24 @@ const GlobalStyle = createGlobalStyle`
...
+    padding: 22px 0 100px;
```
Comment (jkettmann) [inline src/GlobalStyle.js]: This padding is probably very specific to the Home page, right? You can add another styled component to the Home page that [extends the Page styles](https://styled-components.com/docs/basics#extending-styles)

File: src/components/Footer/Footer.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        <FooterLogo />
+      </a>
+      <a className="footer-link" href="/terms-and-privacy">Terms & Privacy</a>
```
Comment (jkettmann) [inline src/components/Footer/Footer.js]: This should be a react-router Link because it handles the history internally. I would suggest you use the react-router link as styled component and for the external link to ooloo.io you can use the [as](https://styled-components.com/docs/api#as-polymorphic-prop) prop to convert the Link into an `a`.

File: src/components/Footer/Footer.js (hunk 3 above)
Comment (jkettmann) [reply src/components/Footer/Footer.js]: And it should be a react-router link as well since it's an internal link

**Review by jkettmann — CHANGES_REQUESTED** [review]
The layout looks good, but I left a couple of comments

**Comment by rotegras** [comment]
I see I could refactor the links in header and footer. It could be much cleaner but since they are so small probably the refactor work would be too much compared to the benefits

**Comment by rotegras** [comment]
And I hadn't seen the as prop. Working on it

File: src/theme/Theme.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    small: '14px',
+    smaller: '13px',
+  },
+  pageContainer: {
+    position: 'relative',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    minHeight: '100vh',
+    padding: '122px 0 100px 0',
```
Comment (jkettmann) [inline src/theme/Theme.js:25]: I guess it would be better to have `padding: 100px 0;` here since that's the height of header and footer. But we can adjust that if necessary on the Search page as well

**Review by jkettmann — CHANGES_REQUESTED** [review]
Looks a lot more consistent. Just the logo in the footer is slightly off. Needs some positive or negative margin-top I guess

image

File: src/App.js (hunk 4 above)
Comment (rotegras) [reply src/App.js]: Yes. it was before finding the problem.

File: src/components/Header/Header.js [hunk 10]
```diff
@@ -3,13 +3,17 @@ import Styled from 'styled-components';
 import NavBar from './NavBar';
 import Logo from './Logo';
 
-
 const StyledHeader = Styled.header`
   display: flex;
   justify-content: space-between;
```
Comment (jkettmann) [inline src/components/Header/Header.js:8]: One thing that I missed during the last review: `justify-content: space-between;` doesn't really work here as you realized by adding the `margin-left` on the logo. An alternative is to set `flex: 1` on the two text links and `text-algin: right` on the right one. Then those take up equal amount of space while the logo only what's required

But I'll leave it up to you to adjust it

**Review by jkettmann — APPROVED** [review]
